Fix flake package

nrdxp opened the pull request:
This isn't ideal for development, since it pulls Helix from GitHub instead of from the underlying filesystem, but it does get the package working for nix users. As an additional convenience, I also embedded the runtime by default.

Opening as a draft as I'm not sure if this is the direction we want to take here, but unless NixOS/nix#3978 get's a proper solution, this may be the best way to do it for now.

Potentially resolves #27

pickfire commented:
Related pull request https://github.com/helix-editor/helix/pull/180

gytis-ivaskevicius commented on flake.nix:
```diff
@@ -6,6 +6,12 @@
     naersk.url = "github:nmattia/naersk";
+    helix = {
+      flake = false;
+      url = "https://github.com/helix-editor/helix";
+      type = "git";
```
Consider replacing it with `url = github:helix-editor/helix;`

nrdxp replied:
Thanks for the review. I actually tried that, as well as `"git:https://..."`, but for some reason it on doesn't understand the submodules attribute when doing so. This is the only way I got it to work.

gytis-ivaskevicius commented on flake.nix:
```diff
@@ -25,7 +31,8 @@
       in rec {
         packages.helix = naerskLib.buildPackage {
           pname = "helix";
-          root = ./.;
+          root = inputs.helix;
```
There really must be some better way? :/

archseer replied:
Needs fixing in Nix :/ https://github.com/NixOS/nix/issues/4423

nrdxp replied:
I did try a `symlinkJoin` and it mostly worked actually. But not if you add new submodules, and it seemed kinda hacky so I left it out.

gytis-ivaskevicius replied:
Wouldn't symlinkJoin allow avoiding `inputs` altogether and would build package from cloned version?

nrdxp replied:
How do you mean?

gytis-ivaskevicius replied:
Ahhh, I see what you did there :D I'd say that it should be left as it is

archseer commented:
I think this looks good, it's not ideal but at least it builds :)

nrdxp commented:
I suppose the only question left then, is if the url should point at master as I currently have it, or to the latest release tag?

nrdxp commented:
I removed the embedded runtime call, since it didn't seem to function as I expected, and I also added an overlay, so users can easily install into a NixOS configuration as well.

archseer commented:
Sidenote: There's also a proposal to nixpkgs (https://github.com/NixOS/nixpkgs/pull/126208) by @yusdacra, I wonder why the tests fail there
